src/core/RefineState.ml

```diff
@@ -10,6 +10,7 @@ type t =
     imports : Namespace.t;
     (** The set of bindings currently in scope. This includes all bindings from the namespaces *)
     (* FIXME: Perhaps we ought to use an array here? We could get O(1) indexing. *)
+    (* FIXME: We should make sure not to bloat the symbol table too much... *)
```

Comment:
What could have been improved?

Reply:
This isn't really the best home for this particular FIXME, nor is it a good description :) 

The issue arises when we import a file repeatedly. In order to get interface files right, we are going to need to go through and normalize all of the symbols defined/used in a given file, and make sure that they are position-independent in a sense. However, once we do that, we won't be able to deduplicate these symbols easily! So really, what we _ought_ to do is build a DAG for our import tree, topsort it, and then go from there. This would prevent this whole "loading the iface files multiple times" issue

Reply:
Well, how about serializing a global symbol as `(unit path, position in that unit)`?

Reply:
This is what I ended up with after fighting with the fancier flat symbol space version for a while.